lib/python/qmk/cli/doctor.py

```diff
@@ -49,23 +49,43 @@ def doctor(cli):
 
     elif OS == "Linux":
         cli.log.info("Detected {fg_cyan}Linux.")
-        if shutil.which('systemctl'):
-            mm_check = subprocess.run(['systemctl', 'list-unit-files'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=10, universal_newlines=True)
-            if mm_check.returncode == 0:
-                mm = False
-                for line in mm_check.stdout.split('\n'):
-                    if 'ModemManager' in line and 'enabled' in line:
-                        mm = True
-
-                if mm:
-                    cli.log.warn("{bg_yellow}Detected ModemManager. Please disable it if you are using a Pro-Micro.")
-
-            else:
-                cli.log.error('{bg_red}Could not run `systemctl list-unit-files`:')
-                cli.log.error(mm_check.stderr)
-
-        else:
-            cli.log.warn("Can't find systemctl to check for ModemManager.")
+        # Checking for udev rules
+        udev_dir = "/etc/udev/rules.d/"
+        # These are the recommended udev rules
+        desired_rules = {"dfu": {'SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2ff4", MODE:="0666"',
+                                 'SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2ffb", MODE:="0666"',
+                                 'SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2ff0", MODE:="0666"'},
+                         "tmk": {'SUBSYSTEMS=="usb", ATTRS{idVendor}=="feed", MODE:="0666"'},
+                         "input-club": {'SUBSYSTEMS=="usb", ATTRS{idVendor}=="1c11", MODE:="0666"'},
+                         "stm32": {'SUBSYSTEMS=="usb", ATTRS{idVendor}=="1eaf", ATTRS{idProduct}=="0003", MODE:="0666"',
+                                   'SUBSYSTEMS=="usb", ATTRS{idVendor}=="0483", ATTRS{idProduct}=="df11", MODE:="0666"'},
+                         "catalina": {'ATTRS{idVendor}=="2a03", ENV{ID_MM_DEVICE_IGNORE}="1"',
```

Comment:
```suggestion
                         "caterina": {'ATTRS{idVendor}=="2a03", ENV{ID_MM_DEVICE_IGNORE}="1"',
```

Reply:
Right, [this line in the docs](https://github.com/qmk/qmk_firmware/blame/master/docs/faq_build.md#L50) throw me off balance.

Reply:
Yeah, that should really be changed too 😄